src/braket/default_simulator/simulation_strategies/batch_operation_strategy.py

```diff
@@ -44,46 +46,90 @@ def apply_operations(
             :math:`(qubit\_count, 0)` tensor
         qubit_count (int): The number of qubits in the state
         operations (list[GateOperation]): The operations to apply to the state vector
-        batch_size: The number of operations to contract in each batch
+        batch_size (int): The number of operations to contract in each batch. Defaults to 1.
 
     Returns:
         np.ndarray: The state vector after applying the given operations, as a type
         (num_qubits, 0) tensor
     """
-    # TODO: Write algorithm to determine partition size based on operations and qubit count
-    partitions = [operations[i : i + batch_size] for i in range(0, len(operations), batch_size)]
+    if not operations:
+        return state
+
+    if batch_size == 1:
+        return _apply_operations_sequential(state, operations)
 
-    for partition in partitions:
-        state = _contract_operations(state, qubit_count, partition)
+    return _apply_operations_batched(state, qubit_count, operations, batch_size)
 
+
+def _apply_operations_sequential(state: np.ndarray, operations: list[GateOperation]) -> np.ndarray:
+    """Apply operations sequentially without batching."""
+    for op in operations:
+        state = _apply_operation(state, op)
     return state
 
 
-def _contract_operations(
-    state: np.ndarray, qubit_count: int, operations: list[GateOperation]
+def _apply_operations_batched(
+    state: np.ndarray, qubit_count: int, operations: list[GateOperation], batch_size: int
 ) -> np.ndarray:
-    contraction_parameters = [state, list(range(qubit_count))]
+    """Apply operations in optimized batches."""
+    current_batch = []
+
+    def process_batch():
+        nonlocal state
```

Comment:
Why the inner function?

Reply:
Originally I had logic in the for loop below. With it removed, I can revert this back out of a function